PurchasesCoreSwift/Networking/Backend.swift

```diff
@@ -0,0 +1,614 @@
+                return
+            }
+
+            guard statusCode < HTTPStatusCodes.redirect.rawValue else {
+                let code = self.maybeNumberFromError(code: response?["code"])
+                let backendMessage = response?["message"] as? String
+                let error = ErrorUtils.backendError(withBackendCode: code, backendMessage: backendMessage)
+                completion(nil, nil, nil, nil, error)
+                return
+            }
+
+            guard let offers = response?["offers"] as? [[String: AnyObject]], offers.count > 0 else {
+                completion(nil, nil, nil, nil, ErrorUtils.unexpectedBackendResponseError())
+                return
+            }
+
+            let offer = offers[0]
+            if let signatureError = offer["signature_error"] as? [String: AnyObject] {
+                let code = self.maybeNumberFromError(code: signatureError["code"])
+                let backendMessage = signatureError["message"] as? String
+                let error = ErrorUtils.backendError(withBackendCode: code, backendMessage: backendMessage)
+                completion(nil, nil, nil, nil, error)
+
+            } else if let signatureData = offer["signature_data"] as? [String: AnyObject] {
+                let signature = signatureData["signature"] as? String
+                let keyIdentifier = offer["key_id"] as? String
+                let nonceString = signatureData["nonce"] as? String
+                let nonce: UUID?
+                if let nonceString = nonceString {
+                    nonce = UUID(uuidString: nonceString)
+                } else {
+                    nonce = nil
+                }
+
+                let timestamp = signatureData["timestamp"] as? NSNumber
+
+                completion(signature, keyIdentifier, nonce, timestamp, nil)
+                return
+            } else {
+                completion(nil, nil, nil, nil, ErrorUtils.unexpectedBackendResponseError())
+                return
+            }
+        }
+    }
+
+    @objc(postAttributionData:network:appUserID:completion:)
+    public func post(attributionData: [String: AnyObject],
+                     network: AttributionNetwork,
+                     appUserID: String,
+                     completion: ((Error?) -> Void)?) {
+        let escapedAppUserID =  escapedAppUserID(appUserID: appUserID)
+        let path = "/subscribers/\(escapedAppUserID)/attribution"
+        let body: [String: Any] = ["network": NSNumber(value: network.rawValue), "data": attributionData]
+        httpClient.performPOSTRequest(serially: true,
+                                      path: path,
+                                      requestBody: body,
+                                      headers: authHeaders) { statusCode, response, error in
+            self.handle(response: response, statusCode: statusCode, error: error, completion: completion)
```

Comment:
[Style] the parameters should be one per line, shouldn't be?

Reply:
Ooops